packages/core/src/VRange.ts

```diff
@@ -95,227 +45,101 @@ export class VRange {
-     * @param [direction] default: Direction.FORWARD
      */
-    setEnd(reference: VNode, position?: RelativePosition, direction = Direction.FORWARD): VRange {
-        if (direction === Direction.FORWARD) {
-            return this._setHead(reference, position);
+    setEnd(reference: VNode, position = RelativePosition.AFTER): this {
+        reference = reference.lastLeaf();
+        if (!reference.hasChildren() && !reference.atomic) {
+            reference.append(this.end);
+        } else if (position === RelativePosition.BEFORE && reference !== this.start) {
+            // We check that `reference` isn't `this.start` to avoid a backward
+            // collapsed range.
+            reference.before(this.end);
         } else {
-            return this._setTail(reference, position);
+            reference.after(this.end);
         }
+        return this;
     }
     /**
-     * Set the start of the range by targetting a `reference` VNode and
-     * specifying the `position` in reference to that VNode ('BEFORE', 'AFTER'),
-     * like in an `xpath`. If no relative position if given, include the
-     * reference node in the selection. Return self.
-     *
-     * @param reference
-     * @param [position]
-     */
-    setAnchor(reference: VNode, position = RelativePosition.BEFORE): VRange {
-        return this._setTail(reference, position);
-    }
-    /**
-     * Set the end of the range by targetting a `reference` VNode and specifying
-     * the `position` in reference to that VNode ('BEFORE', 'AFTER'), like in an
-     * `xpath`. If no relative position if given, include the reference node in
-     * the selection. Return self.
+     * Extend this range in such a way that it includes the given node.
      *
-     * @param reference
-     * @param [position]
-     */
-    setFocus(reference: VNode, position = RelativePosition.AFTER): VRange {
-        return this._setHead(reference, position);
-    }
-    /**
-     * Extend the range from its tail to the given location, targetting a
-     * `reference` VNode and specifying the `direction` of the extension.
+     * This method moves the boundary marker that is closest to the given node
+     * up or down the tree in order to include the given node into the range.
+     * Because of that, calling this method will always result in a range that
+     * is at least the size that it was prior to calling it, and usually bigger.
      *
-     * @param reference
-     * @param [direction] default: Direction.FORWARD
-     * @see _setHead
+     * @param targetNode The node to extend the range to.
      */
-    extendTo(reference: VNode, direction = Direction.FORWARD): void {
+    extendTo(targetNode: VNode): this {
         let position: RelativePosition;
-        if (direction === Direction.FORWARD) {
-            if (reference.hasChildren()) {
-                reference = reference.next();
-                reference = reference.firstLeaf();
+        if (targetNode.isBefore(this.start)) {
```

Comment:
I find this harder to read. How about naming it? `const isExtendingBackward = targetNode.isBefore(this.start);`
Wait... if we're extending to a node that is _before_ the start of the range, we're extending backward indeed. So why was it `direction === Direction.FORWARD` before?

Reply:
It's hard to read from the diff but I actually swapped the `if` and `else` part ! ;-) I'll make a variable for this check.

Reply:
Oh ok now it makes more sense! You're right it's not a backward selection but there is a direction to the extension (left or right if the text is ltr). Because whether that direction is ltr or rtl is dependent on whether the text itself is ltr or rtl, we chose to name these things backward and forward instead. Like `deleteForward` and `deleteBackward`. Nothing to do with the selection either but it's still backward or forward!